---
title: "zebrafish BaP - diversity analyses"
author: "Alexandra Alexiev"
output: md_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
# knit options for whole doc. I will add in specific chunks when I don't want this to be the case
knitr::opts_chunk$set(echo = TRUE, # show code
                      eval = TRUE, # do not run chunks when knitting
                      include = TRUE, # include chunk input in final doc
                      warning = FALSE, # do not show warnings in final doc
                      message = FALSE, # do not show messages from code in final doc
                      collapse = TRUE, # when possible, do put multiple outputs in one block
                      dpi = 300, # fig resolution
                      fig.dim = c(9, 5), # the default figure dimensions
                      out.width = "98%", # the default figure output width
                      out.height = "98%", # the default figure output height 
                      cache = TRUE) # dont rerun chunks that haven't been changed

```

## Housekeeping

```{r libraries and paths}
# libraries
library(ggpubr)
library(dplyr)
library(tidyr)
library(tibble)
library(ggplot2)
library(gridExtra)
library(knitr)
library(stringr)
library(vegan)
library(MASS)
# set seed so stats like permanova are reproducible
set.seed(3)

# to knit an Rmd file, you need explicit file paths typed out, so I created these objects as shortcuts so I could call the object for commonly used file paths instead of retyping the whole thing each time
home_dir <- "/Users/alexieva/Documents/Projects/Analysis/metagen_zfBaP/02_finalAnalysis/"
input_dir <- "/Users/alexieva/Documents/Projects/Analysis/metagen_zfBaP/02_finalAnalysis/input_files_pub/"
output_dir <- "/Users/alexieva/Documents/Projects/Analysis/metagen_zfBaP/02_finalAnalysis/output_files_pub/"

```

```{r read in files whole dataset}
## whole dataset (all generations) files

# presence/absence of pathways in whole dataset
pathcov_tab_ALL <- read.csv(file = paste0(input_dir, "pathcov_tab_ALL.txt"), 
                           sep = "\t",
                           header = TRUE)

# abundance of pathways in whole dataset
pathabund_tab_ALL <- read.csv(file = paste0(input_dir, "pathabund_tab_ALL.txt"), 
                             sep = "\t",
                             header = TRUE)

# metadata for whole dataset
metaALL <- read.csv(file = paste0(input_dir, "meta_tab_ALL.txt"),
                   sep = "\t",
                   header = TRUE) %>%
  mutate(Exposure = relevel(factor(Exposure), "DMSO"), Morpholino = relevel(factor(Morpholino), "CoMo"),
         Generation = factor(Generation),
         Sex = factor(Sex), # relevel and factor so these are accurate
         BMI = Weight_g / (Length_mm/1000)^2) # add BMI calculation
# reorder factors for treatment
metaALL$Treatments <- factor(x = metaALL$Treatments,
                             levels = c("CoMo_DMSO",
                                        "CoMo_BaP",
                                        "AHR2Mo_DMSO",
                                        "AHR2Mo_BaP"))

```

```{r read in files F0}
## F0 files

# presence/absence of pathways in F0
pathcov_tab_F0 <- read.csv(file = paste0(input_dir, "pathcov_tab_F0.txt"), 
                           sep = "\t",
                           header = TRUE)

# abundance of pathways in F0
pathabund_tab_F0 <- read.csv(file = paste0(input_dir, "pathabund_tab_F0.txt"), 
                             sep = "\t",
                             header = TRUE)

# metadata for F0
metaF0 <- read.csv(file = paste0(input_dir, "meta_tab_F0.txt"),
                   sep = "\t",
                   header = TRUE) %>%
  mutate(Exposure = relevel(factor(Exposure), "DMSO"), Morpholino = relevel(factor(Morpholino), "CoMo"),
         Generation = factor(Generation),
         Sex = factor(Sex), # relevel and factor so these are accurate
         BMI = Weight_g / (Length_mm/1000)^2) # add BMI calculation

metaF0$Treatments <- factor(x = metaF0$Treatments,
                             levels = c("CoMo_DMSO",
                                        "CoMo_BaP",
                                        "AHR2Mo_DMSO",
                                        "AHR2Mo_BaP"))

```

```{r read in files F1}
## F1 files

# presence/absence of pathways in F1
pathcov_tab_F1 <- read.csv(file = paste0(input_dir, "pathcov_tab_F1.txt"), 
                           sep = "\t",
                           header = TRUE)

# abundance of pathways in F1
pathabund_tab_F1 <- read.csv(file = paste0(input_dir, "pathabund_tab_F1.txt"), 
                             sep = "\t",
                             header = TRUE)

# metadata for F1
metaF1 <- read.csv(file = paste0(input_dir, "meta_tab_F1.txt"),
                   sep = "\t",
                   header = TRUE) %>%
  mutate(Exposure = relevel(factor(Exposure), "DMSO"), Morpholino = relevel(factor(Morpholino), "CoMo"),
         Generation = factor(Generation),
         Sex = factor(Sex), # relevel and factor so these are accurate
         BMI = Weight_g / (Length_mm/1000)^2) # add BMI calculation

metaF1$Treatments <- factor(x = metaF1$Treatments,
                             levels = c("CoMo_DMSO",
                                        "CoMo_BaP",
                                        "AHR2Mo_DMSO",
                                        "AHR2Mo_BaP"))

```

```{r read in files F2}
## F2 files

# presence/absence of pathways in F2
pathcov_tab_F2 <- read.csv(file = paste0(input_dir, "pathcov_tab_F2.txt"), 
                           sep = "\t",
                           header = TRUE)

# abundance of pathways in F2
pathabund_tab_F2 <- read.csv(file = paste0(input_dir, "pathabund_tab_F2.txt"), 
                             sep = "\t",
                             header = TRUE)

# metadata for F2
metaF2 <- read.csv(file = paste0(input_dir, "meta_tab_F2.txt"),
                   sep = "\t",
                   header = TRUE) %>%
  mutate(Exposure = relevel(factor(Exposure), "DMSO"), Morpholino = relevel(factor(Morpholino), "CoMo"),
         Generation = factor(Generation),
         Sex = factor(Sex), # relevel and factor so these are accurate
         BMI = Weight_g / (Length_mm/1000)^2) # add BMI calculation

metaF2$Treatments <- factor(x = metaF2$Treatments,
                             levels = c("CoMo_DMSO",
                                        "CoMo_BaP",
                                        "AHR2Mo_DMSO",
                                        "AHR2Mo_BaP"))

```


## Alpha diversity stats and plots

# Whole dataset
Calculate alpha diversity metrics (richness, shannon, simpson) on whole dataset (this means all generations worth of data). This modeling approach uses "Generation" as a term, along with other covariates like Sex and Treatment, in the linear models of different alpha diversity metrics, with the input being the whole dataset. AIC is used to determine the optimal model terms.

```{r calc alpha diversity metrics}
pw_richness <- rowSums(pathcov_tab_ALL)
pw_shannon <- diversity(pathabund_tab_ALL, "shannon", base = exp(1))
pw_simpson <- diversity(pathabund_tab_ALL, "simpson")
meta_modALL <- metaALL %>%
  dplyr::select(c("SampleID", "Exposure", "Morpholino", "Generation", "Sex", "Treatments", "BMI")) %>% # filter to only important variables we are testing
  mutate(Treatments = factor(Treatments))
alphadiv <- data.frame(pw_richness, pw_shannon, pw_simpson) %>%
  rownames_to_column("SampleID") %>%
  inner_join(meta_modALL, by = "SampleID")
alphadiv

```

What covariates predict microbiome richness?
```{r linear model richness with covariates}
# step AIC to test which model
testmod_rich <- lm(pw_richness ~ Generation + Exposure + Morpholino + Sex + BMI + Generation:Exposure + Generation:Morpholino + Exposure:Morpholino, data = alphadiv)
AIC_wholedata <- stepAIC(testmod_rich)
# extract lm that is best model for results
lm_rich <- summary(AIC_wholedata)
lm_rich

# best model is with Exposure + Morpholino + Sex + Exposure:Morpholino but Sex (and nothing else) is barely not significant

```

What covarites predict shannon richness?
```{r linear model shannon with covariates}
# step AIC to test which model
testmod_shan <- lm(pw_shannon ~ Generation + Exposure + Morpholino + Sex + BMI + Generation:Exposure + Generation:Morpholino + Exposure:Morpholino, data = alphadiv)
AIC_wholedata <- stepAIC(testmod_shan)

# best model is Generation + Exposure + Morpholino + Sex + Generation:Exposure + Generation:Morpholino + Exposure:Morpholino
# GenerationF1, GenerationF2, GenerationF1:ExposureBaP, GenerationF2:MorpholinoAHR2Mo, ExposureBaP:MorpholinoAHR2Mo significant
# extract lm that is best model for results

lm_shan <-summary(AIC_wholedata) 
lm_shan

```

```{r pathway richness plot}
# richness
pw_richplot <- ggplot(alphadiv, aes(x = Generation, y = pw_richness, 
                                    fill = Treatments)) + 
  geom_boxplot() + 
  theme_classic() +
  labs(x = NULL, y = "Pathway Richness")
pw_richplot

```

```{r intergenerational shannon graph for manuscript}
# shannon
alphadiv$Treatments <- factor(alphadiv$Treatments, 
                                levels = c("CoMo_DMSO",
                                           "CoMo_BaP", 
                                           "AHR2Mo_DMSO",
                                           "AHR2Mo_BaP"))
pw_shanplotint <- ggplot(alphadiv, aes(x = Treatments, y = pw_shannon,
                                    fill = Generation)) + 
  geom_boxplot() +
  theme_classic() +
  labs(x = NULL, y = "Pathway Shannon Diversity") +
  scale_fill_brewer(palette = "PuBu") +
  scale_x_discrete(labels = c("AhR2Mo - / BaP -",
                               "AhR2Mo - / BaP +",
                               "AhR2Mo + / BaP -",
                               "AhR2Mo + / BaP +")) +
  theme(text = element_text(size = 25))

pw_shanplotint


```


The following sections use a subset of the data that is only for each generation of animals. It uses a similar modeling approach as above but without Generation as a variable since one generation at a time is evaluated.

# F0

```{r calc alpha diversity metrics F0}
## calculate alpha diversity metrics and put into a data frame
pwF0_richness <- rowSums(pathcov_tab_F0)
pwF0_shannon <- diversity(pathabund_tab_F0, "shannon", base = exp(1))
pwF0_simpson <- diversity(pathabund_tab_F0, "simpson")
meta_modF0 <- metaF0 %>%
  dplyr::select(c("SampleID", "Exposure", "Morpholino", "Sex", "BMI", "Treatments")) %>% # filter to only important variables we are testing
  mutate(Treatments = factor(Treatments))
alphadiv_pathF0 <- data.frame(pwF0_richness, pwF0_shannon, pwF0_simpson) %>%
  rownames_to_column("SampleID") %>%
  inner_join(meta_modF0, by = "SampleID")
alphadiv_pathF0

```

What covariates predict F1 gut metagenome richness?
```{r linear model richness with covariates F0}
# step AIC to test which model
testmod_richF0 <- lm(pwF0_richness ~ Exposure + Morpholino + Sex + BMI + Exposure:Morpholino + Sex:Exposure + Sex:Morpholino + Sex:BMI, 
                      data = alphadiv_pathF0)
AIC_F0 <- stepAIC(testmod_richF0)

# best model is Exposure + Morpholino + Sex + Exposure:Morpholino + Morpholino:Sex and Sex significant

# save object
lm_richF0 <- summary(AIC_F0)
lm_richF0

# test just sex + treatment + interaction
testmod_rich2F0 <- lm(pwF0_richness ~ Treatments + Sex + BMI + Sex:Treatments + Sex:BMI, 
                      data = alphadiv_pathF0)
summary(stepAIC(testmod_rich2F0))

# Sex best model but not significant

```

What covariates predict F0 gut metagenome shannon diversity?
```{r linear model shannon with covariates F0}
# step AIC to test which model
testmod_shanF0 <- lm(pwF0_shannon ~ Exposure + Morpholino + Sex + BMI + Sex:Exposure + Sex:Morpholino + Exposure:Morpholino + Sex:BMI, 
                    data = alphadiv_pathF0)
AIC_F0 <- stepAIC(testmod_shanF0)

# best model is Morpholino + Sex + Morpholino:Sex
# none significant but some close

lm_shanF0 <- summary(AIC_F0)
lm_shanF0

# test just sex + treatment + interaction
testmod_shan2F0 <- lm(pwF0_shannon ~ Sex + Treatments + BMI + Sex:Treatments, 
                      data = alphadiv_pathF0)
summary(stepAIC(testmod_shan2F0))

# best model is Sex + Treatments + Sex:Treatments
# none significant

plot(lm(formula = pwF0_shannon ~ Sex + Treatments + Sex:Treatments, data = alphadiv_pathF0))
# linear
# normal for the most part, not as much in the lower quantile
# homoskedastic
# no outliers

```

```{r F0 alpha diversity plotting, eval=T}
pwF0_richplot <- ggplot(alphadiv_pathF0, aes(x = Treatments, y = pwF0_richness)) + 
  geom_boxplot() + 
  theme_classic() +
  labs(x = NULL, y = "Pathway Richness") +
  scale_y_continuous(limits = c(20, 55))

pwF0_shanplot <- ggplot(alphadiv_pathF0, aes(x = Treatments, y = pwF0_shannon)) + 
  geom_boxplot() + 
  theme_classic() +
  labs(x = NULL, y = "Shannon Diversity Metric") +
  scale_y_continuous(limits = c(4.2, 4.8))

pwF0_alphaplot <- ggarrange(pwF0_richplot, pwF0_shanplot,
                        labels = c("A", "B"),
                        ncol = 1, nrow = 2)
pwF0_alphaplot

```

# F1

```{r calc alpha diversity metrics F1}
## calculate alpha diversity metrics and put into a data frame
pwF1_richness <- rowSums(pathcov_tab_F1)
pwF1_shannon <- diversity(pathabund_tab_F1, "shannon", base = exp(1))
pwF1_simpson <- diversity(pathabund_tab_F1, "simpson")
meta_modF1 <- metaF1 %>%
  dplyr::select(c("SampleID", "Exposure", "Morpholino", "Sex", "BMI", "Treatments")) %>% # filter to only important variables we are testing
  mutate(Treatments = factor(Treatments))
alphadiv_pathF1 <- data.frame(pwF1_richness, pwF1_shannon, pwF1_simpson) %>%
  rownames_to_column("SampleID") %>%
  inner_join(meta_modF1, by = "SampleID")
alphadiv_pathF1

```

What covariates predict F1 gut metagenome richness?
```{r linear model richness with covariates F1}
# step AIC to test which model
testmod_richF1 <- lm(pwF1_richness ~ Exposure + Morpholino + Sex + BMI + Exposure:Morpholino + Sex:Exposure + Sex:Morpholino + Sex:BMI, 
                      data = alphadiv_pathF1)
AIC_F1 <- stepAIC(testmod_richF1)

# best model is Exposure + Morpholino + Sex + Exposure:Morpholino but not significant (has periods tho)

# save object
lm_richF1 <- summary(AIC_F1)
lm_richF1

# test just sex + treatment + interaction
testmod_rich2F1 <- lm(pwF1_richness ~ Treatments + Sex + BMI + Sex:Treatments, 
                      data = alphadiv_pathF1)
summary(stepAIC(testmod_rich2F1))
# sex best model but insigificant

```

What covariates predict F1 gut metagenome shannon diversity?
```{r linear model shannon with covariates F1}
# step AIC to test which model
testmod_shanF1 <- lm(pwF1_shannon ~ Exposure + Morpholino + Sex + BMI + Sex:Exposure + Sex:Morpholino + Exposure:Morpholino + Sex:BMI, 
                    data = alphadiv_pathF1)
AIC_F1 <- stepAIC(testmod_shanF1)
lm_shanF1 <- summary(AIC_F1)
lm_shanF1

# best model is Exposure + Morpholino + Exposure:Morpholino
# ExposureBaP significant

# test just sex + treatment + interaction
testmod_shan2F1 <- lm(pwF1_shannon ~ Sex + Treatments + BMI + Sex:Treatments, 
                      data = alphadiv_pathF1)
summary(stepAIC(testmod_shan2F1))

# best model is Treatments
# TreatmentsCoMo_BaP significant

plot(lm(formula = pwF1_shannon ~ Treatments, data = alphadiv_pathF1))
# linear
# normal
# homoskedastic
# no outliers

## combine, print to console, and save
lm_alphasF1 <- cbind(lm_richF1, lm_shanF1)

```


```{r F1 alpha diversity plotting, eval=T}
pwF1_richplot <- ggplot(alphadiv_pathF1, aes(x = Treatments, y = pwF1_richness)) + 
  geom_boxplot() + 
  theme_classic() +
  labs(x = NULL, y = "Pathway Richness") + 
  scale_y_continuous(limits = c(20, 55))

pwF1_shanplot <- ggplot(alphadiv_pathF1, aes(x = Treatments, y = pwF1_shannon)) + 
  geom_boxplot() + 
  theme_classic() +
  labs(x = NULL, y = "Shannon Diversity Metric") +
  scale_y_continuous(limits = c(4.2, 4.8))

pwF1_alphaplot <- ggarrange(pwF1_richplot, pwF1_shanplot,
                        labels = c("A", "B"),
                        ncol = 1, nrow = 2)
pwF1_alphaplot

```


# F2

```{r calc alpha diversity metrics F2}
## calculate alpha diversity metrics and put into a data frame
pwF2_richness <- rowSums(pathcov_tab_F2)
pwF2_shannon <- diversity(pathabund_tab_F2, "shannon", base = exp(1))
pwF2_simpson <- diversity(pathabund_tab_F2, "simpson")
meta_modF2 <- metaF2 %>%
  dplyr::select(c("SampleID", "Exposure", "Morpholino", "Sex", "BMI", "Treatments")) %>% # filter to only important variables we are testing
  mutate(Treatments = factor(Treatments))
alphadiv_pathF2 <- data.frame(pwF2_richness, pwF2_shannon, pwF2_simpson) %>%
  rownames_to_column("SampleID") %>%
  inner_join(meta_modF2, by = "SampleID")
alphadiv_pathF2

```

What covariates predict F2 gut metagenome richness?
```{r linear model richness with covariates F2}
# step AIC to test which model
testmod_richF2 <- lm(pwF2_richness ~ Exposure + Morpholino + Sex + BMI + Exposure:Morpholino + Sex:Exposure + Sex:Morpholino + Sex:BMI, 
                      data = alphadiv_pathF2)
AIC_F2 <- stepAIC(testmod_richF2)
# save object
lm_richF2 <- summary(AIC_F2)
lm_richF2

# best model is Morpholino + Sex + Morpholino:Sex with MorpholinoAHR2Mo:SexM significant

# test just sex + treatment + interaction
testmod_rich2F2 <- lm(pwF2_richness ~ Treatments + Sex + BMI + Sex:Treatments, 
                      data = alphadiv_pathF2)
summary(stepAIC(testmod_rich2F2))

# Treatments + Sex + Treatments:Sex best model but none significant

```

What covariates predict F2 gut metagenome shannon diversity?
```{r linear model shannon with covariates F2}
# step AIC to test which model
testmod_shanF2 <- lm(pwF2_shannon ~ Exposure + Morpholino + Sex + BMI + Sex:Exposure + Sex:Morpholino + Exposure:Morpholino + Sex:BMI, 
                    data = alphadiv_pathF2)
AIC_F2 <- stepAIC(testmod_shanF2)
lm_shanF2 <- summary(AIC_F2)
lm_shanF2

# best model is Exposure + Morpholino + Exposure:Morpholino
# ExposureBaP, MorpholinoAHR2Mo, ExposureBaP:MorpholinoAHR2Mo significant

# test just sex + treatment + interaction
testmod_shan2F2 <- lm(pwF2_shannon ~ Sex + Treatments + BMI + Sex:Treatments, 
                      data = alphadiv_pathF2)
summary(stepAIC(testmod_shan2F2))

# best model is Treatments
# TreatmentsCoMo_DMSO significant

plot(lm(formula = pwF2_shannon ~ Treatments, data = alphadiv_pathF2))
# linear
# normal
# homoskedastic
# no outliers

## combine, print to console, and save
lm_alphasF2 <- cbind(lm_richF2, lm_shanF2)

```

```{r F2 alpha diversity plotting, eval=T}
pwF2_richplot <- ggplot(alphadiv_pathF2, aes(x = Treatments, y = pwF2_richness)) + 
  geom_boxplot() + 
  theme_classic() +
  labs(x = NULL, y = "Pathway Richness") +
  scale_y_continuous(limits = c(20, 55))

pwF2_shanplot <- ggplot(alphadiv_pathF2, aes(x = Treatments, y = pwF2_shannon)) + 
  geom_boxplot() + 
  theme_classic() +
  labs(x = NULL, y = "Shannon Diversity Metric") +
  scale_y_continuous(limits = c(4.2, 4.8))

pwF2_alphaplot <- ggarrange(pwF2_richplot, pwF2_shanplot,
                        labels = c("A", "B"),
                        ncol = 1, nrow = 2)
pwF2_alphaplot

```


## Beta diversity stats and plots
- Used distance based RDA with bray curtis distance to create ordination.
- Exploratory analysis revealed that bray curtis, jaccard, and sorenson data frame were correlated significantly (mantel test), meaning that the distance/dissimilarity measure does not change the information and ultimate interpretation downstream. We chose to proceed with bray curtis.
- Ordistep used to choose the optimal model for dbRDA.
- This optimal model was then used in the PERMANOVA to get significance.
- Beta dispersion was also calculated and evaluated with linear regressions and graphed on a box and whisker plot.

# Whole data

Is beta diversity and dispersion of the gut metagenome associated with generation, exposure, morpholino, or other variables?

```{r ordistep model selection, eval=F}
## check model with all possible options (have checked sex with ordistep in the past and doesn't come out important to model)
meta_modALL <- metaALL %>%
  dplyr::select(c("Exposure", "Morpholino", "Generation", "Sex", "BMI"))
ALLmod0 <- capscale(pathabund_tab_ALL ~ 1, meta_modALL, distance = "bray")  # Model with intercept only
ALLmod1 <- capscale(pathabund_tab_ALL ~ . + Exposure:Morpholino + Morpholino:Exposure + Generation:Exposure + Generation:Morpholino + Sex:BMI, meta_modALL, distance = "bray")  # Model with all explanatory variables
pw_ordiALL <- ordistep(ALLmod0, scope = formula(ALLmod1)) # this determines what the best model is to run RDA on
pw_ordiALL

# best model is pathabund_tab_ALL ~ Generation + Exposure + Morpholino + Generation:Exposure + Exposure:Morpholino + Generation:Morpholino
# model inertia was 5.4988; Inertia is scaled Chi-square
# proportion variance explained is 0.2874

```

```{r dbRDA run with best model from ordistep}
# calculate RDA (distance based)
pw_rda <- capscale(formula = pathabund_tab_ALL ~ Generation + Exposure + Morpholino + Generation:Exposure + Exposure:Morpholino + Generation:Morpholino, 
                   data = meta_modALL, distance = "bray")
# using bray because euclidean, bray, and jaccard were all correlated via mantel test, and bray is consistent with other analyses in NMDS and other graphs in this analysis
RsquareAdj(pw_rda)
# 0.2718067 of variation explained by this model

# and with generation as condition just to compare
pw_rda2 <- capscale(formula = pathabund_tab_ALL ~ Condition(Generation) + Exposure + Morpholino + Exposure:Morpholino, 
                   data = meta_modALL, distance = "bray")

```

```{r graph ordination}
# create files for graphing
smry_rda <- summary(pw_rda)

# this is the coordinates for the points and metadata
pw_PC1  <- data.frame(smry_rda$sites[,1:2]) %>%  # these are the x, y coordinates for the sample points (e.g. BaP_####)
  rownames_to_column("SampleID") %>%
  inner_join(dplyr::select(metaALL, c(SampleID, Treatments, Generation, Exposure, Morpholino)), by = "SampleID") %>%
  mutate(treatgen = paste0(Treatments, "_", Generation)) %>%
  column_to_rownames("SampleID")
pw_PC1$Treatments <- factor(x = pw_PC1$Treatments,
                             levels = c("CoMo_DMSO",
                                        "CoMo_BaP",
                                        "AHR2Mo_DMSO",
                                        "AHR2Mo_BaP"))
mylims <- range(with(pw_PC1, c(CAP1, CAP2)))

# the biplot scores are the correlations between your environmental variables and axes
pw_PC2  <- data.frame(smry_rda$biplot)

# put plot together
pwrda_plot <- ggplot(pw_PC1, aes(x = CAP1, y = CAP2)) + 
  geom_point(aes(color = Treatments), size = 2) +
  stat_ellipse(aes(group = treatgen, 
                   color = Treatments), 
               show.legend = T) +
  theme_classic() +
  labs(x = paste0("CAP1 (",round(100*smry_rda$cont$importance[2, "CAP1"], digits = 2),"%)"),
       y = paste0("CAP2 (",round(100*smry_rda$cont$importance[2, "CAP2"], digits = 2),"%)")) +
  scale_color_brewer(palette = "PuOr", direction = -1,
                     labels = c("AhR2Mo - / BaP -",
                               "AhR2Mo - / BaP +",
                               "AhR2Mo + / BaP -",
                               "AhR2Mo + / BaP +")) +
  theme(text = element_text(size = 20)) +
  scale_shape_discrete(name = "Generation shape") +
  scale_linetype_discrete(name = "Ellipse line type") +
  facet_wrap("Generation")

pwrda_plot

```

```{r permanova best model}
pw_dm <- vegdist(pathabund_tab_ALL, method = "bray")
PermExpandMod_pw <- adonis2(pw_dm ~ Generation + Exposure + Morpholino + Generation:Exposure + Exposure:Morpholino + Generation:Morpholino, data = metaALL)
PermExpandMod_pw

```

Beta dipsersion results
```{r beta dispersion calculation and lm stats}
# calc dispersion for each generation/treatment group
metaALL <- metaALL %>% mutate(treatgen = paste0(Treatments, "_", Generation))
pw.disper <- betadisper(pw_dm, metaALL$treatgen)

# make file for graphing
pw.Disper1 <- data.frame(pw.disper$distances)
colnames(pw.Disper1) <- "dispers"
pw.Disper2 <- pw.Disper1 %>%
  rownames_to_column("SampleID") %>%
  inner_join(y = metaALL, 
             by = "SampleID")

## linear model approach
# step AIC to test model
testmod_bdisp <- lm(pw.disper$distances ~ Generation + Exposure + Morpholino + Generation:Exposure + Generation:Morpholino + Exposure:Morpholino, data = pw.Disper2)
AIC_bdispALL <- stepAIC(testmod_bdisp)
# save in object
lm_betadisper <- summary(AIC_bdispALL)
lm_betadisper

# best model was Generation + Exposure + Morpholino + Generation:Exposure + Generation:Morpholino
# GenerationF1:ExposureBaP and GenerationF1:MorpholinoAHR2Mo were only significant ones


```

```{r beta dispersion graph}
pw.disper_bars <- ggplot(data = pw.Disper2, aes(x = Generation, y = log(dispers),
                                                fill = Treatments)) +
  geom_boxplot(outlier.shape = NA) + 
  theme_classic() +
  ylab("log(Dispersion)") + xlab("") +
  geom_point(size = 3, shape = 21, alpha = 0.9,
             position = position_jitterdodge(jitter.width = 0.07)) +
  labs(fill = "Treatments") +
  scale_fill_brewer(palette = "PuOr", direction = -1,
                    labels = c("AhR2Mo - / BaP -",
                               "AhR2Mo - / BaP +",
                               "AhR2Mo + / BaP -",
                               "AhR2Mo + / BaP +")) +
  theme(text = element_text(size = 20))

pw.disper_bars


```


How does treatment comparison change over generation?

Look at spread of variation of beta diversity distance (intergenerational beta diversity analysis) using the median distance between all points in F0, F1, or F2 cluster on ordination and the F0 centroid.

```{r treatment comparison change over generation}
# read in the script to calculate this metric
source("/Users/alexieva/Documents/Projects/Writing/Research manuscripts/metagen_zfBaP/Pub_analysis_gitlab/scripts/treatment_beta_div_across_ALLgens.R")

horiz_treatboxes2 <- ggplot(data = big_data, aes(x = Generation, 
                                                y = median, 
                                                color = Treatments,
                                                group = Treatments)) +
    geom_errorbar(width = 0.1, 
                aes(ymin = median-sd, ymax = median+sd),
                position = position_dodge(0.1)) +
  geom_point(size = 3, position = position_dodge(0.1)) +
  geom_line(position = position_dodge(0.1)) +
  labs(y = "Median dissimilarity\nfrom F0 generation",
       x = "") +
  theme_classic() +
  scale_color_brewer(palette = "PuOr", direction = -1,
                    labels = c("AhR2Mo - / BaP -",
                               "AhR2Mo - / BaP +",
                               "AhR2Mo + / BaP -",
                               "AhR2Mo + / BaP +")) +
  theme(text = element_text(size = 20)) +
  scale_x_discrete(labels = c("F0 (Baseline)", "F1", "F2"))

horiz_treatboxes2


```


The following sections show beta diversity analysis on each individual generation (i.e., subsetting out one generation and running models/ordinations to see trends within each, indpendently of each other). Much like above, for each generation, we are trying to see if beta diversity metrics vary with treatment variables.

# F0

Is beta diversity and dispersion of the gut metagenome of F0 zebrafish associated with exposure, morpholino, or other variables?
```{r ordistep model selection F0, eval=F}
meta_modF0 <- metaF0 %>%
  dplyr::select(c("Exposure", "Morpholino", "Sex")) # filter to only important variables we are testing

F0mod0 <- capscale(pathabund_tab_F0 ~ 1, meta_modF0, distance = "bray")  # Model with intercept only
F0mod1 <- capscale(pathabund_tab_F0 ~ . + Exposure:Morpholino + Morpholino:Exposure, meta_modF0, distance = "bray")  # Model with all explanatory variables
pw_ordiF0 <- ordistep(F0mod0, scope = formula(F0mod1)) # this determines what the best model is to run RDA on
pw_ordiF0

# best model is pathabund_tab_F0 ~ Morpholino + Exposure + Morpholino:Exposure
# model inertia was 1.5837; Inertia is scaled Chi-square
# proportion variance explained is 0.2204

```

```{r dbRDA run with best model from ordistep F0}
# calculate RDA (distance based)
pw_rdaF0 <- capscale(formula = pathabund_tab_F0 ~ Morpholino + Exposure + Morpholino:Exposure, data = meta_modF0, distance = "bray")
# using bray because euclidean, bray, and jaccard were all correlated via mantel test, and bray is consistent with other analyses in NMDS and other graphs in this analysis
RsquareAdj(pw_rdaF0)
# 0.2036827 of variation explained by this model
plot(pw_rdaF0)

```

```{r graph ordination F0}
smry_rdaF0 <- summary(pw_rdaF0)
pwF0_PC1  <- data.frame(smry_rdaF0$sites[,1:2]) %>%      # these are the x, y coordinates for the sample points (e.g. BaP_####)
  rownames_to_column("SampleID") %>%
  inner_join(dplyr::select(metaF0, c(SampleID, Treatments, Exposure, Morpholino)), by = "SampleID") %>%
  column_to_rownames("SampleID")
# the biplot scores are the correlations between your environmental variables and axes
pwF0_PC2  <- data.frame(smry_rdaF0$biplot)     # x and y coordinates for the vectors
pwF0_PC1$Treatments <- factor(pwF0_PC1$Treatments, 
                                levels = c("CoMo_DMSO",
                                           "CoMo_BaP", 
                                           "AHR2Mo_DMSO",
                                           "AHR2Mo_BaP"))

pwrda_plotF0 <- ggplot(pwF0_PC1, aes(x = CAP1, y = CAP2)) + 
  geom_point(aes(color = Treatments), size = 2) +
  theme_classic() +
  geom_hline(yintercept = 0, linetype = "dotted") +
  geom_vline(xintercept = 0, linetype = "dotted") +
  geom_segment(data = pwF0_PC2, aes(x = 0, xend = CAP1, y = 0, yend = CAP2), 
               color = "black", arrow = arrow(length = unit(0.01, "npc"))) +
  geom_text(data = pwF0_PC2, 
            aes(x = CAP1, y = CAP2, label = c("AHR2 Morpholino", "BaP Exposure",
                                              "Interaction")), 
            color = "black", size = 4, hjust = c(0.45, 0, 0.45), vjust = 1) +
  labs(x = paste0("CAP1 (",round(100*smry_rdaF0$cont$importance[2, "CAP1"], digits = 2),"%)"),
       y = paste0("CAP2 (",round(100*smry_rdaF0$cont$importance[2, "CAP2"], digits = 2),"%)")) +  
  scale_color_brewer(palette = "PuOr", direction = -1,
                    name = "Treatments", 
                    labels = c("AhR2Mo - / BaP -",
                               "AhR2Mo - / BaP +",
                               "AhR2Mo + / BaP -",
                               "AhR2Mo + / BaP +")) +
  theme(text = element_text(size = 20))

pwrda_plotF0


```

```{r F0 permanova with best model Morpholino + Exposure + Morpholino:Exposure}
pw_dmF0 <- vegdist(pathabund_tab_F0, method = "bray")
PermExpandMod_pwF0 <- adonis2(pw_dmF0 ~ Morpholino + Exposure + Morpholino:Exposure, data = metaF0)
PermExpandMod_pwF0
# morph > exp > morph:exposure


```

Evaluate beta dispersion at F0 as well.
```{r F0 beta dispersion calculation and lm stats}
# calc dispersion
pw.disperF0 <- betadisper(pw_dmF0, metaF0$Treatments)

# make file for graphing and stats
pwF0.Disper1 <- data.frame(pw.disperF0$distances)
colnames(pwF0.Disper1) <- "dispers"
pwF0.Disper2 <- pwF0.Disper1 %>%
  rownames_to_column("SampleID") %>%
  inner_join(y = metaF0, 
             by = "SampleID")

## linear model approach
# step AIC to test model
testmod_bdispF0 <- lm(pw.disperF0$distances ~ Exposure + Morpholino + Exposure:Morpholino, data = pwF0.Disper2)
AIC_bdispF0 <- stepAIC(testmod_bdispF0)
# save in object
lm_betadisperF0 <- summary(AIC_bdispF0)
lm_betadisperF0
# Morpholino best model and MorpholinoAHR2Mo almost significant (nothing else)

```

```{r F0 beta dispersion graph}
pw.disper_barsF0 <- ggplot(data = pwF0.Disper2, aes(x = Treatments, 
                                          y = dispers)) +
  geom_boxplot(aes(fill = Treatments), alpha = 0.6, 
               outlier.shape = NA) + 
  theme_classic() +
  ylab("Dispersion") + xlab("") +
  geom_point(aes(fill = Treatments), size = 3, shape = 21, position = position_jitter(
    width = 0.25
  )) +
  labs(fill = "Treatments")

pw.disper_barsF0

```


# F1

Is beta diversity and dispersion of the gut metagenome of F1 zebrafish associated with exposure, morpholino, or other variables?
```{r ordistep model selection F1, eval=F}
meta_modF1 <- metaF1 %>%
  dplyr::select(c("Exposure", "Morpholino", "Sex", "BMI")) # filter to only important variables we are testing

F1mod0 <- capscale(pathabund_tab_F1 ~ 1, meta_modF1, distance = "bray")  # Model with intercept only
F1mod1 <- capscale(pathabund_tab_F1 ~ . + Exposure:Morpholino + Morpholino:Exposure + Sex:BMI, meta_modF1, distance = "bray")  # Model with all explanatory variables
pw_ordiF1 <- ordistep(F1mod0, scope = formula(F1mod1)) # this determines what the best model is to run RDA on
pw_ordiF1

# best model is pathabund_tab_F1 ~ Morpholino + Exposure
# model inertia was 2.25210; Inertia is scaled Chi-square
# proportion variance explained is 0.20189

```

```{r dbRDA run with best model from ordistep F1}
# calculate RDA (distance based)
pw_rdaF1 <- capscale(formula = pathabund_tab_F1 ~ Morpholino + Exposure, data = meta_modF1, distance = "bray")
# using bray because euclidean, bray, and jaccard were all correlated via mantel test, and bray is consistent with other analyses in NMDS and other graphs in this analysis
RsquareAdj(pw_rdaF1)
# 0.1901491 of variation explained by this model

```

```{r graph ordination F1}
smry_rdaF1 <- summary(pw_rdaF1)
pwF1_PC1  <- data.frame(smry_rdaF1$sites[,1:2]) %>%      # these are the x, y coordinates for the sample points (e.g. BaP_####)
  rownames_to_column("SampleID") %>%
  inner_join(dplyr::select(metaF1, c(SampleID, Treatments, Exposure, Morpholino)), by = "SampleID") %>%
  column_to_rownames("SampleID")
# the biplot scores are the correlations between your environmental variables and axes
pwF1_PC2  <- data.frame(smry_rdaF1$biplot)     # x and y coordinates for the vectors
pwrda_plotF1 <- ggplot(pwF1_PC1, aes(x = CAP1, y = CAP2)) + 
  geom_point(aes(color = Treatments), size = 2) +
  theme_classic() +
  geom_hline(yintercept = 0, linetype = "dotted") +
  geom_vline(xintercept = 0, linetype = "dotted") +
  geom_segment(data = pwF1_PC2, aes(x = 0, xend = CAP1, y = 0, yend = CAP2), 
               color = "blue", arrow = arrow(length = unit(0.01, "npc"))) +
  geom_text(data = pwF1_PC2, 
            aes(x = CAP1, y = CAP2, label = c(rownames(pwF1_PC2))), 
            color = "black") +
  labs(x = paste0("CAP1 (",round(100*smry_rdaF1$cont$importance[2, "CAP1"], digits = 2),"%)"),
       y = paste0("CAP2 (",round(100*smry_rdaF1$cont$importance[2, "CAP2"], digits = 2),"%)"))

# add centroids to ordination
metaF1_treat <- dplyr::select(metaF1, c("SampleID", "Treatments")) # select only these for join in next step
centDistrda_pwF1 <- as.data.frame(smry_rdaF1$sites) %>% # take only sites rda data
  dplyr::select(c("CAP1", "CAP2")) %>% # select axis points for rda ordination
  rownames_to_column("SampleID") %>% 
  inner_join(metaF1_treat, by = "SampleID") %>% # join with Treatment metadata
  group_by(Treatments) %>% # group Treatments
  summarize(CentroidCAP1 = mean(CAP1), # calculate mean of each treatment's axes
            CentroidCAP2 = mean(CAP2)) %>%
  ungroup() %>% # ungroup
  as.data.frame()
# check if these are in the right place on rda
pwrda_plot_centrF1 <- pwrda_plotF1 + geom_point(data = centDistrda_pwF1,
                          aes(x = CentroidCAP1, y = CentroidCAP2,
                              fill = Treatments),
                          shape = 21, 
                          size = 3, 
                          color = "black",
                          stroke = 1)
pwrda_plot_centrF1

```

```{r F1 permanova with best model Morpholino + Exposure}
pw_dmF1 <- vegdist(pathabund_tab_F1, method = "bray")
PermExpandMod_pwF1 <- adonis2(pw_dmF1 ~ Morpholino + Exposure, data = metaF1)
PermExpandMod_pwF1
# morph > exp


```

Evaluate beta dispersion for F1 fish.
```{r F1 beta dispersion calculation and lm stats}
# calc dispersion
pw.disperF1 <- betadisper(pw_dmF1, metaF1$Treatments)

# make file for graphing
pwF1.Disper1 <- data.frame(pw.disperF1$distances)
colnames(pwF1.Disper1) <- "dispers"
pwF1.Disper2 <- pwF1.Disper1 %>%
  rownames_to_column("SampleID") %>%
  inner_join(y = metaF1, 
             by = "SampleID")

## linear model approach
# step AIC to test model
testmod_bdispF1 <- lm(pw.disperF1$distances ~ Exposure + Morpholino + Exposure:Morpholino, data = pwF1.Disper2)
AIC_bdispF1 <- stepAIC(testmod_bdispF1)
# save in object
lm_betadisperF1 <- summary(AIC_bdispF1)
lm_betadisperF1
# Morpholino best model and MorpholinoAHR2Mo almost significant (nothing else)

```

```{r F1 beta dispersion graph}
pw.disper_barsF1 <- ggplot(data = pwF1.Disper2, aes(x = Treatments, 
                                          y = dispers)) +
  geom_boxplot(aes(fill = Treatments), alpha = 0.6, 
               outlier.shape = NA) + 
  theme_classic() +
  ylab("Dispersion") + xlab("") +
  geom_point(aes(fill = Treatments), size = 3, shape = 21, position = position_jitter(
    width = 0.25
  )) +
  labs(fill = "Treatments")

pw.disper_barsF1

```


# F2

Is beta diversity and dispersion of the gut metagenome of F2 zebrafish associated with exposure, morpholino, or other variables?
```{r ordistep model selection F2, eval=F}
meta_modF2 <- metaF2 %>%
  dplyr::select(c("Exposure", "Morpholino", "Sex", "BMI")) # filter to only important variables we are testing

F2mod0 <- capscale(pathabund_tab_F2 ~ 1, meta_modF2, distance = "bray")  # Model with intercept only
F2mod1 <- capscale(pathabund_tab_F2 ~ . + Exposure:Morpholino + Morpholino:Exposure + Sex:BMI, meta_modF2, distance = "bray")  # Model with all explanatory variables
pw_ordiF2 <- ordistep(F2mod0, scope = formula(F2mod1)) # this determines what the best model is to run RDA on
pw_ordiF2

# best model is pathabund_tab_F2 ~ Morpholino + Exposure + Morpholino:Exposure
# model inertia was 1.5020; Inertia is scaled Chi-square
# proportion variance explained is 0.2722

```

```{r dbRDA run with best model from ordistep F2}
# calculate RDA (distance based)
pw_rdaF2 <- capscale(formula = pathabund_tab_F2 ~ Morpholino + Exposure + Morpholino:Exposure, data = meta_modF2, distance = "bray")
# using bray because euclidean, bray, and jaccard were all correlated via mantel test, and bray is consistent with other analyses in NMDS and other graphs in this analysis
RsquareAdj(pw_rdaF2)
# 0.25603 of variation explained by this model

```

```{r graph ordination F2}
smry_rdaF2 <- summary(pw_rdaF2)
pwF2_PC1  <- data.frame(smry_rdaF2$sites[,1:2]) %>%      # these are the x, y coordinates for the sample points (e.g. BaP_####)
  rownames_to_column("SampleID") %>%
  inner_join(dplyr::select(metaF2, c(SampleID, Treatments, Exposure, Morpholino)), by = "SampleID") %>%
  column_to_rownames("SampleID")
# the biplot scores are the correlations between your environmental variables and axes
pwF2_PC2  <- data.frame(smry_rdaF2$biplot)     # x and y coordinates for the vectors
pwrda_plotF2 <- ggplot(pwF2_PC1, aes(x = CAP1, y = CAP2)) + 
  geom_point(aes(color = Treatments), size = 2) +
  theme_classic() +
  geom_hline(yintercept = 0, linetype = "dotted") +
  geom_vline(xintercept = 0, linetype = "dotted") +
  geom_segment(data = pwF2_PC2, aes(x = 0, xend = CAP1, y = 0, yend = CAP2), 
               color = "blue", arrow = arrow(length = unit(0.01, "npc"))) +
  geom_text(data = pwF2_PC2, 
            aes(x = CAP1, y = CAP2, label = c(rownames(pwF2_PC2))), 
            color = "black") +
  labs(x = paste0("CAP1 (",round(100*smry_rdaF2$cont$importance[2, "CAP1"], digits = 2),"%)"),
       y = paste0("CAP2 (",round(100*smry_rdaF2$cont$importance[2, "CAP2"], digits = 2),"%)"))

# add centroids to ordination
metaF2_treat <- dplyr::select(metaF2, c("SampleID", "Treatments")) # select only these for join in next step
centDistrda_pwF2 <- as.data.frame(smry_rdaF2$sites) %>% # take only sites rda data
  dplyr::select(c("CAP1", "CAP2")) %>% # select axis points for rda ordination
  rownames_to_column("SampleID") %>% 
  inner_join(metaF2_treat, by = "SampleID") %>% # join with Treatment metadata
  group_by(Treatments) %>% # group Treatments
  summarize(CentroidCAP1 = mean(CAP1), # calculate mean of each treatment's axes
            CentroidCAP2 = mean(CAP2)) %>%
  ungroup() %>% # ungroup
  as.data.frame()
# check if these are in the right place on rda
pwrda_plot_centrF2 <- pwrda_plotF2 + geom_point(data = centDistrda_pwF2,
                          aes(x = CentroidCAP1, y = CentroidCAP2,
                              fill = Treatments),
                          shape = 21, 
                          size = 3, 
                          color = "black",
                          stroke = 1)
pwrda_plot_centrF2

```

```{r F2 permanova with best model Morpholino + Exposure + Morpholino:Exposure}
pw_dmF2 <- vegdist(pathabund_tab_F2, method = "bray")
PermExpandMod_pwF2 <- adonis2(pw_dmF2 ~ Morpholino + Exposure + Morpholino:Exposure, data = metaF2)
PermExpandMod_pwF2
# morph > exp > interaction


```

Evaluate beta dispersion for F2 fish.
```{r F2 beta dispersion calculation and lm stats}
# calc dispersion
pw.disperF2 <- betadisper(pw_dmF2, metaF2$Treatments)

# make file for graphing
pwF2.Disper1 <- data.frame(pw.disperF2$distances)
colnames(pwF2.Disper1) <- "dispers"
pwF2.Disper2 <- pwF2.Disper1 %>%
  rownames_to_column("SampleID") %>%
  inner_join(y = metaF2, 
             by = "SampleID")

## linear model approach
# step AIC to test model
testmod_bdispF2 <- lm(pw.disperF2$distances ~ Exposure + Morpholino + Exposure:Morpholino, data = pwF2.Disper2)
AIC_bdispF2 <- stepAIC(testmod_bdispF2)
# save in object
lm_betadisperF2 <- summary(AIC_bdispF2)
lm_betadisperF2
# Exposure + Morpholino best model and MorpholinoAHR2Mo almost significant (nothing else)

```

```{r F2 beta dispersion graph}
pw.disper_barsF2 <- ggplot(data = pwF2.Disper2, aes(x = Treatments, 
                                          y = dispers)) +
  geom_boxplot(aes(fill = Treatments), alpha = 0.6, 
               outlier.shape = NA) + 
  theme_classic() +
  ylab("Dispersion") + xlab("") +
  geom_point(aes(fill = Treatments), size = 3, shape = 21, position = position_jitter(
    width = 0.25
  )) +
  labs(fill = "Treatments")
pw.disper_barsF2


```


## Various useful figures

Paneled beta dispersion figure of each generation's dispersion (most were not significant or were very low impact on trends)
```{r paneled beta dispersion fig}
pw.disper_barsF0 <- pw.disper_barsF0 + 
  scale_y_continuous(limits = c(0, 0.7))
pw.disper_barsF1 <- pw.disper_barsF1 + 
  scale_y_continuous(limits = c(0, 0.7))
pw.disper_barsF2 <- pw.disper_barsF2 + 
  scale_y_continuous(limits = c(0, 0.7))
pw_disperbars_bygen <- ggarrange(pw.disper_barsF0, pw.disper_barsF1, pw.disper_barsF2, ncol = 3, nrow = 1)
pw_disperbars_bygen


```

dbRDA of each generation in a paneled supplemental figure, based on the optimal models for each subset generation.
```{r manuscript suppl figure of each individual generation rda}
########################### F0 ########################### 
# pre plot data frame
smry_rdaF0 <- summary(capscale(formula = pathabund_tab_F0 ~ Morpholino + Exposure + Morpholino:Exposure, 
                               data = meta_modF0,
                               distance = "bray"))
F0_PC1  <- data.frame(smry_rdaF0$sites[,1:2]) %>%      # these are the x, y coordinates for the sample points (e.g. BaP_####)
  rownames_to_column("SampleID") %>%
  inner_join(dplyr::select(metaF0, c(SampleID, Treatments, Exposure, Morpholino)), by = "SampleID") %>%
  column_to_rownames("SampleID")
F0_PC2  <- data.frame(smry_rdaF0$biplot)     # x and y coordinates for the vectors
# reorder x-axis
F0_PC1$Treatments <- factor(F0_PC1$Treatments, 
                        levels = c("CoMo_DMSO",
                                   "CoMo_BaP", 
                                   "AHR2Mo_DMSO",
                                   "AHR2Mo_BaP"))

rda_plotF0 <- ggplot(F0_PC1, aes(x = CAP1, y = CAP2)) + 
  geom_point(aes(color = Treatments), size = 2) +
  theme_classic() +
  geom_hline(yintercept = 0, linetype = "dotted") +
  geom_vline(xintercept = 0, linetype = "dotted") +
  geom_segment(data = F0_PC2, aes(x = 0, xend = CAP1, y = 0, yend = CAP2), 
               color = "black", arrow = arrow(length = unit(0.01, "npc"))) +
  geom_text(data = F0_PC2, 
            aes(x = CAP1, y = CAP2, label = c("AHR2 Morpholino", "BaP Exposure", "Interaction")), 
            color = "black", size = 4, hjust = 0.5, vjust = 0.5) +
  labs(x = paste0("CAP1 (",round(100*smry_rdaF0$cont$importance[2, "CAP1"], digits = 2),"%)"),
       y = paste0("CAP2 (",round(100*smry_rdaF0$cont$importance[2, "CAP2"], digits = 2),"%)")) +  
  scale_color_brewer(palette = "PuOr", direction = -1,
                     name = "Treatments",
                     labels = c("AHR2Mo - / BaP -",
                               "AHR2Mo - / BaP +",
                               "AHR2Mo + / BaP -",
                               "AHR2Mo + / BaP +")) +
  theme(text = element_text(size = 20))

rda_plotF0


########################### F1 ########################### 
# pre plot data frame
smry_rdaF1 <- summary(capscale(formula = pathabund_tab_F1 ~ Morpholino + Exposure, 
                               data = meta_modF1,
                               distance = "bray"))
F1_PC1  <- data.frame(smry_rdaF1$sites[,1:2]) %>%      # these are the x, y coordinates for the sample points (e.g. BaP_####)
  rownames_to_column("SampleID") %>%
  inner_join(dplyr::select(metaF1, c(SampleID, Treatments, Exposure, Morpholino)), by = "SampleID") %>%
  column_to_rownames("SampleID")
F1_PC2  <- data.frame(smry_rdaF1$biplot)     # x and y coordinates for the vectors
# reorder x-axis
F1_PC1$Treatments <- factor(F1_PC1$Treatments, 
                        levels = c("CoMo_DMSO",
                                   "CoMo_BaP", 
                                   "AHR2Mo_DMSO",
                                   "AHR2Mo_BaP"))

rda_plotF1 <- ggplot(F1_PC1, aes(x = CAP1, y = CAP2)) + 
  geom_point(aes(color = Treatments), size = 2) +
  theme_classic() +
  geom_hline(yintercept = 0, linetype = "dotted") +
  geom_vline(xintercept = 0, linetype = "dotted") +
  geom_segment(data = F1_PC2, aes(x = 0, xend = CAP1, y = 0, yend = CAP2), 
               color = "black", arrow = arrow(length = unit(0.01, "npc"))) +
  geom_text(data = F1_PC2, 
            aes(x = CAP1, y = CAP2, label = c("AHR2 Morpholino", "BaP Exposure")), 
            color = "black", size = 4, hjust = 1, vjust = 0.5) +
  labs(x = paste0("CAP1 (",round(100*smry_rdaF1$cont$importance[2, "CAP1"], digits = 2),"%)"),
       y = paste0("CAP2 (",round(100*smry_rdaF1$cont$importance[2, "CAP2"], digits = 2),"%)")) +  
  scale_color_brewer(palette = "PuOr", direction = -1,
                     name = "Treatments",
                     labels = c("AHR2Mo - / BaP -",
                               "AHR2Mo - / BaP +",
                               "AHR2Mo + / BaP -",
                               "AHR2Mo + / BaP +")) +
  scale_shape_discrete(labels = c("Female", "Male")) +
  theme(text = element_text(size = 20))

rda_plotF1


########################### F2 ########################### 
# pre plot data frame
smry_rdaF2 <- summary(capscale(formula = pathabund_tab_F2 ~ Morpholino + Exposure + Morpholino:Exposure, 
                               data = meta_modF2,
                               distance = "bray"))
F2_PC1  <- data.frame(smry_rdaF2$sites[,1:2]) %>%      # these are the x, y coordinates for the sample points (e.g. BaP_####)
  rownames_to_column("SampleID") %>%
  inner_join(dplyr::select(metaF2, c(SampleID, Treatments, Exposure, Morpholino)), by = "SampleID") %>%
  column_to_rownames("SampleID")
F2_PC2  <- data.frame(smry_rdaF2$biplot)     # x and y coordinates for the vectors
# reorder x-axis
F2_PC1$Treatments <- factor(F2_PC1$Treatments, 
                        levels = c("CoMo_DMSO",
                                   "CoMo_BaP", 
                                   "AHR2Mo_DMSO",
                                   "AHR2Mo_BaP"))

rda_plotF2 <- ggplot(F2_PC1, aes(x = CAP1, y = CAP2)) + 
  geom_point(aes(color = Treatments), size = 2) +
  theme_classic() +
  geom_hline(yintercept = 0, linetype = "dotted") +
  geom_vline(xintercept = 0, linetype = "dotted") +
  geom_segment(data = F2_PC2, aes(x = 0, xend = CAP1, y = 0, yend = CAP2), 
               color = "black", arrow = arrow(length = unit(0.01, "npc"))) +
  geom_text(data = F2_PC2, 
            aes(x = CAP1, y = CAP2, label = c("AHR2 Morpholino", "BaP Exposure", "Interaction")), 
            color = "black", size = 4, hjust = c(0.5, 0, 0), vjust = 0.5) +
  labs(x = paste0("CAP1 (",round(100*smry_rdaF2$cont$importance[2, "CAP1"], digits = 2),"%)"),
       y = paste0("CAP2 (",round(100*smry_rdaF2$cont$importance[2, "CAP2"], digits = 2),"%)")) +  
  scale_color_brewer(palette = "PuOr", direction = -1,
                     name = "Treatments",
                     labels = c("AHR2Mo - / BaP -",
                               "AHR2Mo - / BaP +",
                               "AHR2Mo + / BaP -",
                               "AHR2Mo + / BaP +")) +
  scale_shape_discrete(labels = c("Female", "Male")) +
  theme(text = element_text(size = 20))

rda_plotF2


```

Heatmap of all stats - behavior, alpha, and beta diversity - per treatment
```{r paneled heatmap of all stats}
# read in data and clean up for graphing
sumstats_beh <- read.csv(paste0(input_dir, "SumStatsHeat_behavior.csv"),
                         header = T) %>%
  mutate(Treatment = replace(Treatment, Treatment == "CoMo _BaP", "AHR2Mo - / BaP +"),
         Treatment = replace(Treatment, Treatment == "AHRMo_DMSO", "AHR2Mo + / BaP -"),
         Treatment = replace(Treatment, Treatment == "AHRMo_BaP", "AHR2Mo + / BaP +"),
         Metric = replace(Metric, Metric == "freeswim", "Freeswim"),
         Metric = replace(Metric, Metric == "shoaling_iid", "Shoaling iid"),
         Metric = replace(Metric, Metric == "shoaling_nnd", "Shoaling nnd"),
         Metric = replace(Metric, Metric == "shoaling_speed", "Shoaling speed"))
sumstats_beh$Metric <- factor(x = sumstats_beh$Metric,
                             levels = c("Freeswim",
                                        "Shoaling nnd",
                                        "Shoaling iid",
                                        "Shoaling speed"))

sumstats_alphadiv <- read.csv(paste0(input_dir, "SumStatsHeat_alphadiv.csv"),
                         header = T) %>%
  mutate(Treatment = replace(Treatment, Treatment == "CoMo _BaP", "AHR2Mo - / BaP +"),
         Treatment = replace(Treatment, Treatment == "AHRMo_DMSO", "AHR2Mo + / BaP -"),
         Treatment = replace(Treatment, Treatment == "AHRMo_BaP", "AHR2Mo + / BaP +"),
         Metric = replace(Metric, Metric == "Richness_16S", "Taxonomic Richness"),
         Metric = replace(Metric, Metric == "Shannon_16S", "Taxonomic Shannon Diversity"),
         Metric = replace(Metric, Metric == "Richness_MG", "Pathway Richness"),
         Metric = replace(Metric, Metric == "Shannon_MG", "Pathway Shannon Diversity"))
sumstats_alphadiv$Metric <- factor(x = sumstats_alphadiv$Metric,
                             levels = c("Taxonomic Richness",
                                        "Taxonomic Shannon Diversity",
                                        "Pathway Richness",
                                        "Pathway Shannon Diversity"))

sumstats_betadiv <- read.csv(paste0(input_dir, "SumStatsHeat_betadiv.csv"),
                         header = T) %>%
  mutate(Treatment = replace(Treatment, Treatment == "CoMo _BaP", "AHR2Mo - / BaP +"),
         Treatment = replace(Treatment, Treatment == "AHRMo_DMSO", "AHR2Mo + / BaP -"),
         Treatment = replace(Treatment, Treatment == "AHRMo_BaP", "AHR2Mo + / BaP +"),
         Metric = replace(Metric, Metric == "BC_16S", "Taxonomic Bray-Curtis Dissimilarity"),
         Metric = replace(Metric, Metric == "BC_MG", "Pathway Bray-Curtis Dissimilarity"))
sumstats_betadiv$Metric <- factor(x = sumstats_betadiv$Metric,
                             levels = c("Taxonomic Bray-Curtis Dissimilarity",
                                        "Pathway Bray-Curtis Dissimilarity"))

# make a heatmap per measure
hm_stats_beh <- ggplot(data = sumstats_beh, aes(x = Metric, y = Treatment)) +
  geom_tile(aes(fill = Statistic), colour = "white", linetype = 1) +
  theme_classic() +
  scale_fill_gradient2(low = "white", high = "darkgreen", 
                       na.value = "white")  + 
  labs(y = "Treatment compared to Control", x = "", fill = "KS test\nStatistic") +
  geom_text(aes(label = Significance)) + 
  coord_fixed() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1),
        strip.text.y.right = element_text(angle = 0),
        text = element_text(size = 23),
        plot.margin = unit(c(0,0.2,0,1), 'lines')) +
  facet_grid("Generation") +
  ggtitle("Behavioral metrics")

hm_stats_alpha <- ggplot(data = sumstats_alphadiv, aes(x = Metric, y = Treatment)) +
  geom_tile(aes(fill = Estimate), colour = "white", linetype = 1) +
  theme_classic() +
  scale_fill_gradient2(low = "tan", mid = "white", high = "darkgreen", 
                       midpoint = 0, na.value = "white")  + 
  labs(y = "Treatment compared to Control", x = "", fill = "Linear Model\nEstimate") +
  geom_text(aes(label = Significance)) + 
  coord_fixed() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1),
        strip.text.y.right = element_text(angle = 0),
        text = element_text(size = 23),
        plot.margin = unit(c(0,0.2,0,1), 'lines')) +
  facet_grid("Generation") +
  ggtitle("Alpha diversity metrics")

hm_stats_beta <- ggplot(data = sumstats_betadiv, aes(x = Metric, y = Treatment)) +
  geom_tile(aes(fill = Estimate), colour = "white", linetype = 1) +
  theme_classic() +
  scale_fill_gradient2(low = "tan", mid = "white", high = "darkgreen", 
                       midpoint = 0, na.value = "white")  + 
  labs(y = "Treatment compared to Control", x = "", fill = "Linear Model\nEstimate") +
  geom_text(aes(label = Significance)) + 
  coord_fixed() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1),
        strip.text.y.right = element_text(angle = 0),
        text = element_text(size = 23),
        plot.margin = unit(c(0,0.2,0,1), 'lines')) +
  facet_grid("Generation") +
  ggtitle("Beta diversity metrics")

# combine into a paneled fig and save
hm_sumstates <- ggarrange(hm_stats_beh, hm_stats_alpha, hm_stats_beta, 
                          ncol = 3, nrow = 1,
                          align = "hv",
                          labels = c("A", "B", "C"))
hm_sumstates


```



